cmds/core/timeout/main_test.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2021 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bytes"
+	"regexp"
+	"testing"
+
+	"github.com/u-root/u-root/pkg/testutil"
+)
+
+// TestOK for now just runs a simple successful test with 0 args or more than one arg.
+func TestOK(t *testing.T) {
+	var tests = []struct {
+		args   []string
+		stdout string
+		stderr string
+		exitok bool
+	}{
+		{args: []string{}, stdout: "", exitok: false},
+		{args: []string{"date"}, stdout: ".*", exitok: true},
+		{args: []string{"-t", "wh", "date"}, stdout: ".*", stderr: ".*invalid.*duration.*wh", exitok: false},
+		{args: []string{"echo", "hi"}, stdout: ".*hi", exitok: true},
+		{args: []string{"-t", "3s", "sleep", "30s"}, exitok: false},
+		{args: []string{"-v", "echo", "hi"}, stderr: ".*Run.*echo.*", stdout: ".*hi.*", exitok: true},
+	}
+
+	for _, v := range tests {
+		c := testutil.Command(t, v.args...)
+		stdout, stderr := &bytes.Buffer{}, &bytes.Buffer{}
+		c.Stdout, c.Stderr = stdout, stderr
+		err := c.Run()
+		if (err != nil) && v.exitok {
+			t.Errorf("%v: got %v, want nil", v, err)
+		}
+		if (err == nil) && !v.exitok {
+			t.Errorf("%v: got nil, want err", v)
+		}
+		m, err := regexp.MatchString(v.stderr, stderr.String())
+		if err != nil {
+			t.Errorf("stderr: %v: got %v, want nil", v, err)
+		} else {
```

Comment:
you can get rid of this "else" and use t.Fatalf on line 44 (and probably like 36 and 40) if you use subtests  https://go.dev/blog/subtests

for _, t := range tests {
  t.Run(t.name, func(t *testing.T) {
    // rest of existing code here... using t.Fatalf will mark the current test case (subtest) as failed and move on to the next one 
  }
}